civis/ml/_model.py

```diff
@@ -798,15 +812,17 @@ def train(self, df=None, csv_path=None, table_name=None,
 
         if (HAS_SKLEARN and HAS_JOBLIB and
                 isinstance(self.model, BaseEstimator)):
-            with tempfile.TemporaryDirectory() as tempdir:
-                fout = os.path.join(tempdir, 'estimator.pkl')
-                joblib.dump(self.model, fout, compress=3)
-                with open(fout, 'rb') as _fout:
-                    n = self.model_name if self.model_name else "CivisML"
-                    estimator_file_id = cio.file_to_civis(
-                        _fout, 'Estimator for ' + n, client=self._client)
-                self._input_model = self.model  # Keep the estimator
+            tempdir = tempfile.mkdtemp()
+            fout = os.path.join(tempdir, 'estimator.pkl')
+            joblib.dump(self.model, fout, compress=3)
+            with open(fout, 'rb') as _fout:
+                n = self.model_name if self.model_name else "CivisML"
+                estimator_file_id = cio.file_to_civis(
+                    _fout, 'Estimator for ' + n, client=self._client)
+            self._input_model = self.model  # Keep the estimator
             self.model = str(estimator_file_id)
+            os.remove(fout)
+            os.removedirs(tempdir)
```

Comment:
The cleanup should be in a `finally` block so that we clean up temporary files even if there's an exception.